### PR TITLE
Week 12 - Movie Project - Nadia Mazzoni

### PR DESCRIPTION
**Pull request by Bluepoisson**
Link to my deployed project:
https://laughing-ardinghelli-d65cab.netlify.app/

My reflections on how this week's project turned out:
Overall pleased witg the project.

Things I'd like to have clarified or explained in more detail:
useEffect dependencies

### CODE REVIEW

File: code/src/pages/MoviesList.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      });
+  }, [])
+
+  return (
+//map over state to render chosen elements
+    <div className="movies_list">
+      {moviesList.map(({id, poster_path, title, release_date}) => (
+        <div key={id} className="movie_wrapper">
+          <Link key={moviesList.id} to={`movies/${id}`}>
+            <img className="movies_list_img" src={`https://image.tmdb.org/t/p/w342${poster_path}`} alt="movie poster" />
```
Comment (BamStugendal) [inline code/src/pages/MoviesList.js:26]: Here in `alt="movie poster" /> ` you could use `{title}` so the alt says more specific what poster it is and better for accessibility. So for example `alt="{title} movie poster" />` 😄

File: code/src/pages/MoviesList.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+export const MoviesList = () => {
+  const [moviesList, setMoviesList] = useState([]);  // stores moviesList
+
+
+  useEffect(() => {
+    fetch(POPLIST_API) 
+      .then(res => res.json())
+      .then(data => {
+        setMoviesList(data.results) //get list of movies frm api and pute data in state
```
Comment (BamStugendal) [inline code/src/pages/MoviesList.js:16]: Great job at commenting! 👍

File: code/src/pages/NotFound.js [hunk 3]
```diff
@@ -0,0 +1,2 @@
+
+
```
Comment (BamStugendal) [inline code/src/pages/NotFound.js:2]: Delate component if you're not using it :)

File: code/src/pages/movielist.css [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+  margin-top: 4px;
+  margin-left: 2px;
+}
+
+
+
+
+
+
+
```
Comment (BamStugendal) [inline code/src/pages/movielist.css:74]: Empty lines.

File: code/src/pages/moviedetails.css [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+    position: absolute;
+    margin-left: 23px;
+    margin-top: 18px;
+    z-index: 5;
+
+}
+
+.backLink:hover svg {
+    margin-right: 5px;
+    transition: margin-right 0.2s cubic-bezier(0.42, 0, 0.21, 0.99);
```
Comment (BamStugendal) [inline code/src/pages/moviedetails.css:34]: Cool transition! 😄

**Review by BamStugendal — APPROVED** [review]
Great job Nadia! Looks identical to the Technigo movie site! And good commenting on the code, making it easy for me to understand the code 😄

File: code/src/App.js [hunk 6]
```diff
@@ -1,9 +1,31 @@
...
+    {/* enables switches between the different routes and displaying only one */}
+      <Switch>
+        {/* Wrap the comp in a path  */}
+      <Route path="/" exact>
+      <MoviesList />
+      </Route>
+      <Route path="/movies/:movieId" exact>
+      <MovieDetails />
+      </Route>
+      {/* <Route path="/404">
```
Comment (EmOlsz) [inline code/src/App.js:24]: Wrong indentations + commented code. Please, fix that

File: code/src/pages/MovieDetails.js [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+      .then(res => res.json())
+      .then(data => {
+        setMovieDetails(data)
+    });
+  }, [URL_DETAILS, movieId]);
+  // }, [URL_DETAILS,movieId]) it asked specififically for the url dependendencie as movieId array?
+
+  // Ensure we have details to paint. If not, do not paint anything. I could have loading hereIn order to avoid "404 error undefined" in console during fetch.
+  if (!movieDetails.backdrop_path) {
+    return null;
```
Comment (EmOlsz) [inline code/src/pages/MovieDetails.js:28]: Great job :)

File: code/src/pages/MovieDetails.js [hunk 8]
```diff
@@ -0,0 +1,58 @@
...
+  }
+
+    return (
+      <article className="details_page">
+        <Route path='/movies'>
```
Comment (EmOlsz) [inline code/src/pages/MovieDetails.js:33]: Hmm, you don't need `<Route />` over here, let's remove it

File: code/src/pages/MoviesList.js (hunk 1 above)
Comment (Bluepoisson) [reply code/src/pages/MoviesList.js:26]: Yes! Thanks :)

File: code/src/pages/MovieDetails.js (hunk 8 above)
Comment (Bluepoisson) [reply code/src/pages/MovieDetails.js]: I can't remember what I was trying to do, may have confused it with Link. 
Removed also in import.

File: code/src/pages/MovieDetails.js (hunk 7 above)
Comment (Bluepoisson) [reply code/src/pages/MovieDetails.js:28]: Thanks :)

File: code/src/pages/moviedetails.css (hunk 5 above)
Comment (Bluepoisson) [reply code/src/pages/moviedetails.css:33]: Thanx :)

File: code/src/pages/movielist.css (hunk 4 above)
Comment (Bluepoisson) [reply code/src/pages/movielist.css:74]: Good catch! Removed :)

File: code/src/pages/NotFound.js (hunk 3 above)
Comment (Bluepoisson) [reply code/src/pages/NotFound.js]: Absolutely, thought I was going to use it but nooo. Gone :)

**Pull request closed without merging**